release v1.0.3 (#1130)
pingcap/tidb-operator: CHANGELOG.md

# TiDB Operator v1.0.3 Release Notes

## v1.0.3 What's New

### Action Required

This release adds the `timezone` support for [all charts](https://github.com/pingcap/tidb-operator/tree/master/charts).

> For existing tidb clusters. If the `timezone` in `tidb-cluster/values.yaml` has been customized to other timezones instead of the default `UTC`, then upgrading tidb-operator will trigger a rolling update for the related pods.

Reviewer: ```suggestion
For existing TiDB clusters. If the `timezone` in `tidb-cluster/values.yaml` has been customized to other timezones instead of the default `UTC`, then upgrading tidb-operator will trigger a rolling update for the related pods.
```

# TiDB Operator v1.0.3 Release Notes

## v1.0.3 What's New

### Action Required

This release adds `timezone` support for [all charts](https://github.com/pingcap/tidb-operator/tree/master/charts).

> For the `tidb-cluster` chart, we already have the `timezone` option(default `UTC`). If the user does not change it to a different value(for example: `Aisa/Shanghai`), all pods will not be recreated.

Reviewer: ```suggestion
For the `tidb-cluster` chart, the `timezone` option (default `UTC`) is already included. No pod will be recreated if you do not change it to a different value.
```

> If the user changes it to other value(for example: `Aisa/Shanghai`), all the related pods (add a `TZ` env) will be recreated(rolling update).

Reviewer: ```suggestion
If you change it to another value (for example: `Aisa/Shanghai`), all the related pods (add a `TZ` env) will be recreated via a rolling update process.
```

Reviewer: > (add a `TZ` env)

What does this mean?

Author: time zone

> For other charts, we don't have a `timezone` option in their `values.yaml`. We add the `timezone` option in this PR. Whether the user uses the old `values.yaml` or the new `values.yaml`, all the related pods (add a `TZ` env) will not be recreated(rolling update).

Reviewer: ```suggestion
For other charts, there is no `timezone` option in their `values.yaml` files. We add the `timezone` option in this PR. Whether you use the old `values.yaml` or the new `values.yaml`, the related pods (add a `TZ` env) will not be recreated (rolling update).
```

> The related pods include `pump`, `drainer`, `dicovery`, `monitor`, `scheduled backup`, `tidb-initializer`, `tikv-importer`.

Reviewer: ```suggestion
The related pods include `pump`, `drainer`, `dicovery`, `monitor`, `scheduled backup`, `tidb-initializer`, and `tikv-importer`.
```

> All images' time zone maintained by `tidb-operator` are `UTC`. You need to make sure that the time zone inside your images are `UTC` if you use your own images.

Reviewer: ```suggestion
The time zone for all images maintained by `tidb-operator` should be `UTC`. If you use your own images, you need to make sure that the corresponding time zones are `UTC`.
```

### Improvements

> - Add `timezone` support for all containers of tidb cluster

Reviewer: ```suggestion
- Add the `timezone` support for all containers of the TiDB cluster
```

> - Support configuring resources requests and limits for all containers of tidb cluster

Reviewer: ```suggestion
- Support configuring resource requests and limits for all containers of the TiDB cluster
```

## Detailed Bug Fixes and Changes

> - Add `timezone` support for all containers of tidb cluster ([#1122](https://github.com/pingcap/tidb-operator/pull/1122))

Reviewer: ```suggestion
- Add the `timezone` support for all containers of the TiDB cluster ([#1122](https://github.com/pingcap/tidb-operator/pull/1122))
```

> - Support configuring resources requests and limits for all containers of tidb cluster ([#853](https://github.com/pingcap/tidb-operator/pull/853))

Reviewer: ```suggestion
- Support configuring resource requests and limits for all containers of the TiDB cluster ([#853](https://github.com/pingcap/tidb-operator/pull/853))
```
